homeassistant/components/mythicbeastsdns.py

```diff
@@ -0,0 +1,105 @@
+"""
+Integrate with Mythic Beasts Dynamic DNS service.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/mythicbeastsdns/
+"""
+import asyncio
+from datetime import timedelta
+import logging
+
+import aiohttp
+import async_timeout
+import voluptuous as vol
+
+from homeassistant.const import (CONF_HOST, CONF_DOMAIN, CONF_PASSWORD)
+import homeassistant.helpers.config_validation as cv
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = 'mythicbeastsdns'
+
+DEFAULT_INTERVAL = timedelta(minutes=10)
+
+TIMEOUT = 10
+UPDATE_URL = 'https://dnsapi4.mythic-beasts.com/'
+
+CONF_UPDATE_INTERVAL = 'update_interval'
+
+CONFIG_SCHEMA = vol.Schema({
+    DOMAIN: vol.Schema({
+        vol.Required(CONF_HOST, DOMAIN): cv.string,
+        vol.Required(CONF_DOMAIN, DOMAIN): cv.string,
+        vol.Required(CONF_PASSWORD, DOMAIN): cv.string,
```

Comment:
`DOMAIN` is not needed. `vol.Required(CONF_HOST): cv.string,` is enough.

Reply:
I dont think this is correct, the domain is part of the authentication scheme for the mythic beasts dnsapi.  https://www.mythic-beasts.com/support/api/primary

Reply:
I was talking about the [validation](http://alecthomas.github.io/voluptuous/docs/_build/html/voluptuous.html?highlight=required#voluptuous.schema_builder.Required) itself and not the entries.